lego/apps/events/views.py

```diff
@@ -201,3 +204,32 @@ def admin_register(self, request, *args, **kwargs):
             raise APINoSuchPool()
         reg_data = RegistrationReadDetailedSerializer(registration).data
         return Response(data=reg_data, status=status.HTTP_201_CREATED)
+
+
+class RegistrationSearchViewSet(AllowedPermissionsMixin,
+                                mixins.CreateModelMixin,
+                                viewsets.GenericViewSet):
+    serializer_class = RegistrationSearchSerializer
+    ordering = 'registration_date'
+
+    def get_queryset(self):
+        event_id = self.kwargs.get('event_pk', None)
+        return Registration.objects.filter(event=event_id).prefetch_related('user')
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        username = serializer.data['username']
+
+        try:
+            reg = self.get_queryset().get(user__username=username)
+        except Registration.DoesNotExist:
+            raise ValidationError({'error': 'No user with that username'})
+
+        if not get_permission_handler(Event).has_perm(request.user, 'EDIT', obj=reg.event):
+            raise PermissionDenied()
+
+        reg.presence = constants.PRESENT
+        reg.save()
+        data = PublicUserSerializer(reg.user).data
```

Comment:
actually, @martinhath, maybe this should return the updated registration? think that would work better with the frontend and reducers and shit (we have the full name from the search result anyway)

Reply:
Ah, right! Yeah, that sounds like a good idea.